token_dispenser/token_dispenser_lambda.py

```diff
@@ -0,0 +1,175 @@
+from cryptography.hazmat.primitives.serialization.pkcs12 import load_key_and_certificates
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.serialization import Encoding, PrivateFormat, NoEncryption
+from tempfile import NamedTemporaryFile
+from token_dispenser.aws.s3 import download_s3_file
+from token_dispenser.aws.secret_manager import get_secret_value
+from token_dispenser.aws.launchpad_token import get_token
+import token_dispenser.configuration as config
+from token_dispenser.repository.token_repo import put_token, get_token_by_client_id
+import os, json, re
+import logging
+from token_dispenser.logging_config import initialize_logger, shared_logger
+import time
+
+cached_cert_file:NamedTemporaryFile = None
+# Set the logging level dynamically
+log_level = getattr(logging, config.LOG_LEVEL, logging.INFO)
+
+def decode_pkcs12(p12_file_path, password: str):
+    """
+    Decodes a PKCS#12 file to extract the private key, certificate, and additional certificates.
+
+    :param p12_file_path: Path to the PKCS#12 file (.p12 or .pfx).
+    :param password: Password for the PKCS#12 file (if any).
+    :return: A tuple containing the private key, certificate, and additional certificates.
+    """
+    with open(p12_file_path, "rb") as file:
+        pkcs12_data = file.read()
+
+    # Decode the PKCS#12 file
+    private_key, certificate, additional_certs = load_key_and_certificates(
+        pkcs12_data,
+        password.encode("utf-8"),
+        backend=default_backend()
+    )
+    return private_key, certificate, additional_certs
+
+def build_cached_cert_file(private_key, certificate):
+    logger = shared_logger()
+    global cached_cert_file
+    try:
+        logger.debug(f"entered build_cached_cert_file")
+        # Serialize private key and certificate into PEM format
+        private_key_pem = private_key.private_bytes(
+            encoding=Encoding.PEM,
+            format=PrivateFormat.PKCS8,
+            encryption_algorithm=NoEncryption()
+        ).decode("utf-8")
+
+        certificate_pem = certificate.public_bytes(Encoding.PEM).decode("utf-8")
+        # Combine private key and certificate into a single file for mutual TLS authentication
+        # The single file is managed by tempfile.NamedTemporaryFile where the library creates an arbitrary file
+        # and be responsible to remove the file while exiting the python virtual machine even if exception happened
+        if cached_cert_file is None:
+            cached_cert_file = NamedTemporaryFile(delete=True)  # Important: delete=True. deletion after python exits
```

Comment:
`NamedTemporaryFile` already has delete as `True` by default, no need to explicitly add it